src/main/java/org/clas/detectors/DCmonitor.java

```diff
@@ -236,6 +230,8 @@ public void plotHistos() {
     @Override
     public void processEvent(DataEvent event) {
                 
+	if (reverse == null) reverse = getReverseTT(forward);
```

Comment:
reverse needs to be rebuilt if the run number changes because forward can be different. That's why getReverseTT was called in createHistos

Reply:
then need to check run number changed

Reply:
are there other options?

Reply:
createHistos is called at the start, if the run number changes, and when a manual or automatic reset is issued.

if getReverseTT is called in createHistos, it will be executed in the three vases listed above. The first (at the start) is not really useful because the run number is not known yet and could be slept with a check on run number being non-zero

Reply:
we should really avoid blocking the UI for 10+ seconds if feasible